packages/reporter/src/commands/command-model.ts

```diff
@@ -76,6 +76,16 @@ export default class Command extends Instrument {
     this.renderProps = props.renderProps || {}
     this.visible = props.visible
 
+    if (this.timeout !== props.timeout) {
+      const timeElapsed = Date.now() - new Date(this.wallClockStartedAt).getTime()
+      const percentageComplete = timeElapsed / this.timeout
+      const newTimeout = (props.timeout - timeElapsed) / (1 - percentageComplete)
+      const newPosition = percentageComplete * newTimeout - timeElapsed
+
+      this.timeout = newTimeout
+      this.wallClockStartedAt = new Date(new Date(this.wallClockStartedAt).getTime() - newPosition).toJSON()
```

Comment:
Can you put line 80's `new Date(this.wallClockStartedAt).getTime()` in a const and reuse? It would clean up this line a lot more.